### PR TITLE
[fix] load lr scheduler state dict during checkpoint loading

### PR DESCRIPTION
**Pull request by ronghanghu**
Summary: the current implementation does not load lr_scheduler's states from checkpoints, which is a bug (causing last_epoch to be reset when loading from checkpoints). This commit fixes it by saving and loading lr_scheduler's states. It fall backs to setting lr_scheduler's last_epoch when lr_scheduler's states don't exist in checkpoint.

Test Plan: tested with M4C checkpoint saving and loading.

(CPU tests failed due to black formatting error in mmf/datasets/processors/bert_processors.py, unrelated to this PR.)

### CODE REVIEW

File: mmf/trainers/mmf_trainer.py [hunk 1]
```diff
@@ -55,9 +55,12 @@ def configure_callbacks(self):
         self.lr_scheduler_callback = LRSchedulerCallback(self.config, self)
 
         # Add callbacks for execution during events
+        self.callbacks.append(self.lr_scheduler_callback)
```
Comment (apsdehal) [inline mmf/trainers/mmf_trainer.py:58]: Good catch.

**Comment by ronghanghu** [comment]
@apsdehal I rebased onto master and imported it to Phabricator.
